tests/test_dataset.py

```diff
@@ -0,0 +1,296 @@
+    filename = path.join(dirname, "../testoutput/WKDataset")
+    ds = WKDataset.create(filename, [1])
+    ds.add_layer("color", "color")
+
+    ds.get_layer("color").add_mag("1")
+    ds.get_layer("color").add_mag("2-2-1")
+
+    assert path.exists(path.join(dirname, "../testoutput/WKDataset/color/1"))
+    assert path.exists(path.join(dirname, "../testoutput/WKDataset/color/2-2-1"))
+
+    assert len(ds.properties.data_layers) == 1
+    assert len(ds.properties.data_layers["color"].wkw_resolutions) == 2
+
+
+def test_create_TiffDataset_with_layer_and_mag():
+    delete_dir("../testoutput/TiffDataset")
+
+    dirname = path.dirname(__file__)
+    filename = path.join(dirname, "../testoutput/TiffDataset")
+    ds = WKDataset.create(filename, [1])
+    ds.add_layer("color", "color")
+
+    ds.get_layer("color").add_mag("1")
+    ds.get_layer("color").add_mag("2-2-1")
+
+    assert path.exists(path.join(dirname, "../testoutput/TiffDataset/color/1"))
+    assert path.exists(
+        path.join(dirname, "../testoutput/TiffDataset/color/2-2-1")
+    )
+
+    assert len(ds.properties.data_layers) == 1
+    assert len(ds.properties.data_layers["color"].wkw_resolutions) == 2
+
+
+def test_open_wk_dataset():
+    dirname = path.dirname(__file__)
+    filename = path.join(dirname, "../testdata/simple_wk_dataset")
+    ds = WKDataset.open(filename)
+
+    assert len(ds.properties.data_layers) == 1
+    assert len(ds.properties.data_layers["color"].wkw_resolutions) == 1
+
+
+def test_slice_read_with_open():
+    dirname = path.dirname(__file__)
+    mag_filename = path.join(dirname, "../testdata/simple_wk_dataset/color/1")
+    slice = WKSlice(mag_filename)
+
+    assert not slice._is_opened
+
+    with slice.open():
+        assert slice._is_opened
+
+        data = slice.read((10, 10, 10))
+        assert data.shape == (3, 10, 10, 10)  # three channel
+
+    assert not slice._is_opened
+
+
+def test_slice_read_without_open():
+    dirname = path.dirname(__file__)
+    mag_filename = path.join(dirname, "../testdata/simple_wk_dataset/color/1")
+    slice = WKSlice(mag_filename)
+
+    assert not slice._is_opened
+
+    # 'read()' checks if it was already opened. If not, it opens and closes automatically
+    data = slice.read((10, 10, 10))
+    assert data.shape == (3, 10, 10, 10)  # three channel
+
+    assert not slice._is_opened
+
+
+def test_slice_wk_write():
+    dirname = path.dirname(__file__)
+    old_dataset_path = path.join(dirname, "../testdata/simple_wk_dataset/")
+    new_dataset_path = path.join(dirname, "../testoutput/simple_wk_dataset/")
+    mag_filename = path.join(new_dataset_path, "color/1")
+
+    delete_dir(new_dataset_path)
+
+    copytree(old_dataset_path, new_dataset_path)
+
+    slice = WKSlice(mag_filename, size=(100, 100, 100))
+
+    with slice.open():
+        data = slice.read((10, 10, 10))
+        assert not np.array_equal(data, np.zeros((3, 10, 10, 10)))
+
+        slice.write(np.zeros((3, 10, 10, 10), dtype=np.uint8))
```

Comment:
I'd rather generate some random numbers, write these and then compare these against the generated numbers. You can use numpy for that. I also recommend to initialize the seed of numpy random with a static value so that the test is deterministic (e.g., `np.random.seed(1234)`).